Main/ClientApp/src/Infrastructure/FetchUtilities.ts

```diff
@@ -4,25 +4,41 @@ export async function authorizedGet(apiEndpoint: string) {
   const host = window.location.origin;
   const token = await authService.getAccessToken();
   return await fetch(`${host}/${apiEndpoint}`, {
-    headers: !token ? {} : { Authorization: `Bearer ${token}` }
+    headers: !token ? {} : { Authorization: `Bearer ${token}` },
   });
 }
 
 export async function authorizedPost(apiEndpoint: string, data = {}) {
+  return authorizedGenericFetch(apiEndpoint, data, "POST");
+}
+
+export async function authorizedDelete(apiEndpoint: string, data = {}) {
+  return authorizedGenericFetch(apiEndpoint, data, "DELETE");
+}
+
+export async function authorizedUpdate(apiEndpoint: string, data = {}) {
+  return authorizedGenericFetch(apiEndpoint, data, "PUT");
+}
+
+async function authorizedGenericFetch(
```

Comment:
It's not generic, there is no type parameter here. "authorizedRequest" would be a better name